infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/ThymeleafConfigurator.java

```diff
@@ -13,24 +13,31 @@
 
 package tech.pegasys.teku.infrastructure.restapi;
 
-import io.javalin.plugin.rendering.JavalinRenderer;
-import io.javalin.plugin.rendering.template.JavalinThymeleaf;
+import io.javalin.rendering.template.JavalinThymeleaf;
+import java.util.concurrent.atomic.AtomicBoolean;
 import org.thymeleaf.TemplateEngine;
 import org.thymeleaf.templatemode.TemplateMode;
 import org.thymeleaf.templateresolver.ClassLoaderTemplateResolver;
 import org.thymeleaf.templateresolver.ITemplateResolver;
 
-/** ThymeLeaf templates configuration for Javalin */
-public class ThymeleafConfigurator {
+/**
+ * ThymeLeaf templates configuration for Javalin. Note that Javalin renderers are configured using a
+ * singleton so only the first call to this class will have any effect.
+ */
+class ThymeleafConfigurator {
+  private static final AtomicBoolean REGISTERED = new AtomicBoolean();
```

Comment:
Could we have `AtomicBoolean(false)` here, please. It may be obvious with boolean (not for me), but it looks like it even allows `new AtomicInteger()`, which is the same as `new AtomicInteger(0)` and it sounds weird going that far

Reply:
Yeah good point, I like making the default values explicit too.